core/src/main/java/com/google/instrumentation/stats/ContextUtils.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import com.google.instrumentation.common.NonThrowingCloseable;
+import io.grpc.Context;
+
+/**
+ * Util methods/functionality to interact with the {@link io.grpc.Context}.
+ *
+ * <p>Users must interact with the current Context via the public APIs in {@link
+ * StatsContextFactory} and avoid usages of the {@link #STATS_CONTEXT_KEY} directly.
+ */
+public final class ContextUtils {
```

Comment:
> I'd prefer to put STATS_CONTEXT_KEY in another class since all the other methods in ContextUtils are package-private, but this might be inconsistent with trace.

Yeah, it would be nice to keep this part of the API consistent, though I think it is less important than keeping the main API consistent.
/cc @bogdandrutu 

> Or maybe mark STATS_CONTEXT_KEY as Deprecated?

I would rather change the name than deprecate the field, because an element can be deprecated for many reasons, but a name can give the reason that the element's use is discouraged.  Also, if someone has a good reason to use the field, I think it would be better if they didn't need to suppress deprecation warnings.